java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -63,20 +71,62 @@ abstract class AbstractArcHost(vararg initialParticles: ParticleRegistration) :
     // VisibleForTesting
     protected fun getArcHostContext(arcId: String) = runningArcs[arcId]
 
-    /** Subclasses may override this to load persistent context state. */
-    protected open suspend fun lookupOrCreateArcHostContext(
+    protected suspend fun lookupOrCreateArcHostContext(
         partition: Plan.Partition
-    ): ArcHostContext = runningArcs[partition.arcId] ?: ArcHostContext()
+    ): ArcHostContext = lookupArcHostContext(partition.arcId) ?: ArcHostContext()
+
+    protected suspend fun lookupArcHostContext(arcId: String) =
+        runningArcs[arcId] ?: readContextFromStorage(arcId)
 
     /**
      * Called to persist [ArcHostContext] after [context] for [arcId] has been modified.
      *
-     * Subclasses may override this to store the [context] in a more persistent, durable location.
      */
-    protected open suspend fun updateArcHostContext(arcId: String, context: ArcHostContext) {
+    protected suspend fun updateArcHostContext(arcId: String, context: ArcHostContext) {
         runningArcs[arcId] = context
+        writeContextToStorage(arcId, context)
     }
 
+    /**
+     * Creates a specialized [ArcHostContextParticle] used for serializing [ArcHostContext] state
+     * to storage.
+     */
+    private suspend fun createArcHostContextParticle(arcId: String) =
+        ArcHostContextParticle().apply {
+            val partition = createArcHostContextPersistencePlan(
+                arcHostContextCapability,
+                arcId,
+                hostId
+            )
+            partition.particles.get(0).handles.forEach { handleSpec ->
```

Comment:
For future thought -- maybe it'd be better to get the particle by name, from a readability/maintainability perspective. Perhaps we need an XPath-like way to query plans :smile:

Reply:
I can add an accessor to the data classes later. Not a bad idea.